main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -724,7 +727,7 @@ object Weeder2 {
         case Some(expr) => Validation.success(expr)
         case None =>
           // Fall back on Expr.Error. Parser has reported an error here.
-          val err = ParseError("Expected expression", SyntacticContext.Expr.OtherExpr, tree.loc)
+          val err = ParseError(fmt => s"Expected ${fmt.cyan("<expression>")}.", SyntacticContext.Expr.OtherExpr, tree.loc)
```

Comment:
We usually do all the formatting inside the error. 

Would it be possible to break ParseError into several kinds of errors?

Reply:
I agree that it's not nice to have to do formatting in `Parser2`. Let me see if I can come up with a nicer solution!